### ISSUE TITLE
[HttpClient] URL with colon is not correctly parsed

### ISSUE THREAD

**Issue by garak** [symfony/symfony#54332]
### Symfony version(s) affected

6.4 and 7.0

### Description

I'm using Google API for address validation. As referenced in their [documentation](https://developers.google.com/maps/documentation/address-validation/reference/rest#rest-resource:-v1), the URL to call is `/v1:validateAddress`

The problem is that HTTP client is trying to parse the URL without including the domain, so the part before the colon is interpreted as a schema, resulting in a "Unsupported scheme" HTTPException.

### How to reproduce

Define an HTTP client like this:

```yaml
framework:
    http_client:
        scoped_clients:
            map_address.client:
                base_uri: 'https://addressvalidation.googleapis.com/'
                headers:
                    content-type: 'application/json'
                    accept: 'application/json'
```

then inject your client as `HttpClientInterface $mapAddressClient` and try to use it:

```php
$response = $this->mapAddressClient->request('POST', 'v1:validateAddress?key='.$myKey, [/* some data */]);
```
You'll get a `Symfony\Component\HttpKernel\Exception\ HttpException` `Unsupported scheme in "v1:validateAddress?key=<redacted>`

### Possible Solution

I guess that the parsing of the URL done in https://github.com/symfony/symfony/blob/a2fc0924a6118d772f0768e270569aae57ff0411/src/Symfony/Component/HttpClient/HttpClientTrait.php#L621 should be always done using the full URL, including the schema and the domain, so basically including the configured "base_uri" option.

**Comment by MatTheCat** [comment]
`:` is a colon; `,` is a comma :grin: 

According to the RFC 3986 this is the correct implementation (your case is [the first “normal example”](https://datatracker.ietf.org/doc/html/rfc3986#section-5.4.1)).

Prefix your path with `/` to prevent it to be parsed as a scheme.

**Comment by garak** [comment]
The proposed solution works, so I close.
I edited the title/content of the issue, so someone looking for the same problem will be able to find it.
Thanks @MatTheCat

**Comment by atomiix** [comment]
Hi!
I’ve been facing the same issue but adding only a `/` (`/test:1234`) didn’t solve the issue, I got:
```
[MyApi.php] on line 35:
[Symfony\Component\HttpClient\Exception\InvalidArgumentException] {#830 ▼
  #message: "Malformed URL "/test:124"."
  #code: 0
```

However, when I add a `?` at the end of the url in addition to the `/` at the beginning (`/test:1234?`), it works!

I think it should not behave differently depending on the presence or not of a `?`.

**Comment by nicolas-grekas** [comment]
This is due to https://github.com/php/php-src/issues/12703
We might need to work around...

**Comment by nicolas-grekas** [comment]
Support for `/foo:123` should be fixed by #58218
